tls_tests/src/test_clone_read.c

```diff
@@ -0,0 +1,74 @@
+/* 
+ * EC440 Project 4
+ * Thread Local Storage and Copy-on-Write
+ * 
+ * File:   test_clone_read.c
+ * Author: Tony Faller
+ *
+ * Created on April 2, 2021, 2:12 PM
+ */
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <time.h>
+#include <assert.h>
+#include <string.h>
+
+int tls_create(unsigned int size);
+int tls_destroy();
+int tls_read(unsigned int offset, unsigned int length, char *buffer);
+int tls_write(unsigned int offset, unsigned int length, const char *buffer);
+int tls_clone(pthread_t tid);
+
+int ret;
+char *msg = "Hello";
+int pthread_id;
```

Comment:
This should be type `pthread_t`. Depending on the underlying data type of `pthread_t`, storing as an int may result in both width and sign changes when stored as an int and later cast back.